doc/modules/grid_search.rst

```diff
@@ -167,6 +171,358 @@ variable that is log-uniformly distributed between ``1e0`` and ``1e3``::
+    ...                          ratio=2, resource='n_estimators',
+    ...                          max_resources=30).fit(X, y)
+    >>> sh.best_estimator_
+    RandomForestClassifier(max_depth=5, n_estimators=24, random_state=0)
+
+Note that it is not possible to budget on a parameter that is part of the
+parameter grid.
+
+.. _exhausting_the_resources:
+
+Exhausting the available resources
+----------------------------------
+
+As mentioned above, the number of resources that is used at each iteration
+depends on the `min_resources` parameter.
+If you have a lot of resources available but start with a low number of
+resources, some of them might be wasted (i.e. not used)::
+
+    >>> from sklearn.datasets import make_classification
+    >>> from sklearn.svm import SVC
+    >>> from sklearn.experimental import enable_successive_halving  # noqa
+    >>> from sklearn.model_selection import HalvingGridSearchCV
+    >>> import pandas as pd
+    >>> param_grid= {'kernel': ('linear', 'rbf'),
+    ...              'C': [1, 10, 100]}
+    >>> base_estimator = SVC(gamma='scale')
+    >>> X, y = make_classification(n_samples=1000)
+    >>> sh = HalvingGridSearchCV(base_estimator, param_grid, cv=5,
+    ...                          ratio=2, min_resources=20).fit(X, y)
+    >>> sh.n_resources_
+    [20, 40, 80]
+
+The search process will only use 80 resources at most, while our maximum
+amount of available resources is ``n_samples=1000``. Here, we have
+``min_resources = r_0 = 20``.
+
+For :class:`HalvingGridSearchCV`, by default, the `min_resources` parameter
+is set to 'exhaust'. This means that `min_resources` is automatically set
+such that the last iteration can use as many resources as possible, within
+the `max_resources` limit::
+
+    >>> sh = HalvingGridSearchCV(base_estimator, param_grid, cv=5,
+    ...                          ratio=2, min_resources='exhaust').fit(X, y)
+    >>> sh.n_resources_
+    [250, 500, 1000]
+
+`min_resources` was here automatically set to 250, which results in the last
+iteration using all the resources. The exact value that is used depends on
+the number of candidate parameter, on `max_resources` and on `ratio`.
+
+For :class:`HalvingRandomSearchCV`, exhausting the resources can be done in 2
+ways:
+
+- by setting `min_resources='exhaust'`, just like for
+  :class:`HalvingGridSearchCV`;
+- by setting `n_candidates='exhaust'`.
+
+Both options are mutally exclusive: using `min_resources='exhaust'` requires
+knowing the number of candidates, and symmetrically `n_candidates='exhaust'`
+requires knowing `min_resources`.
+
+In general, exhausting the total number of resources leads to a better final
+candidate parameter, and is slightly more time-intensive.
+
+.. _aggressive_elimination:
+
+Aggressive elimination of candidates
+------------------------------------
+
+Ideally, we want the last iteration to evaluate ``ratio`` candidates (see
+:ref:`amount_of_resource_and_number_of_candidates`). We then just have to
```

Comment:
We could always just pick the best one at any point, no matter how many points are left ;)